src/app/loadout-drawer/LoadoutDrawerHeader.tsx

```diff
@@ -11,23 +14,44 @@ export default function LoadoutDrawerHeader({
   loadout: Readonly<Loadout>;
   onNameChanged(name: string): void;
 }) {
-  const setName = (e: React.ChangeEvent<HTMLInputElement>) => onNameChanged(e.target.value);
+  const setName = (e: React.ChangeEvent<HTMLTextAreaElement>) =>
+    onNameChanged(e.target.value.replace(/[\r\n]/g, ''));
+  const inputRef = useRef<HTMLTextAreaElement>(null);
+  const [insertionIndex, setInsertionIndex] = useState<number | null>(null);
+  const emptyTags = useCallback(() => [], []);
+
+  useAutocomplete(inputRef, emptyTags);
 
   return (
     <div className={styles.loadoutName}>
       <ClassIcon classType={loadout.classType} />
-      <input
-        className={styles.dimInput}
-        name="name"
-        onChange={setName}
-        minLength={1}
-        maxLength={50}
-        required={true}
-        autoComplete="off"
-        type="text"
-        value={loadout.name}
-        placeholder={t('Loadouts.LoadoutName')}
-      />
+      <div className={styles.dimInput}>
+        <textarea
+          className="a"
+          ref={inputRef}
+          name="name"
+          onChange={setName}
+          rows={1}
+          minLength={1}
+          maxLength={50}
+          required={true}
+          autoComplete="off"
+          value={loadout.name}
+          onBlur={() => setInsertionIndex(inputRef.current!.selectionStart ?? null)}
+          placeholder={t('Loadouts.LoadoutName')}
+        />
```

Comment:
textcomplete doesn't support <input />, so this makes a <textarea /> behave as much as an <input /> as possible.